backend/resources/center/main.js

```diff
@@ -0,0 +1,88 @@
+//@ts-check
+
+(function () {
+    const vscode = acquireVsCodeApi();
+
+    const oldState = vscode.getState() || { guides: [] };
+
+    /** @type {Array<{ extensionId: string, id: string, title: string, description: string, disabled?: boolean }>} */
+    let guides = oldState.guides;
+    updateGuideList(guides);
+
+    // Handle messages sent from the extension to the webview
+    window.addEventListener('message', event => {
+        const message = event.data;
+        switch (message.type) {
+            case 'refreshData':
+                {
+                    guides = message.data.items;
+                    updateGuideList(guides);
+                    break;
+                }
+        }
+    });
+
+    /**
+     * @param {Array<{ extensionId: string, id: string, title: string, description: string, disabled?: boolean }>} guides
+     */
+    function updateGuideList(guides) {
+        const ul = document.querySelector('.guide-list');
+        const oParent = ul?.parentElement;
+        ul.textContent = '';
```

Comment:
The `ul` could be `Element` or `null` according to the method signature, so this line might also use optional chaining operator `?.` similar to 30th line?

Reply:
Actually if guides or ul is undefined or null, it will return. so the optional operator '?' is not necessary, and I'll remove it.